### PR TITLE
Set required and aria-required on the input when the input is required

### PR DESCRIPTION
**Pull request by CharlieWinkwaves**
https://github.com/bootstrap-ruby/bootstrap_form/issues/663

### CODE REVIEW

**Comment by donv** [comment]
@CharlieWinkwaves Thanks!  Can you fix the two rubocop offences?

File: lib/bootstrap_form/inputs/check_box.rb
Comment (lcreid) [inline lib/bootstrap_form/inputs/check_box.rb:30]: It took me a minute to figure out why we might want to make a check box required. I assume it's for cases where you won't accept the form unless the users checks something like: "I read and agree with the terms."

Would you mind adding a test to `test/bootstrap_checkbox_test.rb` that shows that the aria attribute is added correctly? I think the code is fine, but we need the tests to ensure that we don't introduce regressions in the future.

File: lib/bootstrap_form/form_group_builder.rb
Comment (lcreid) [inline lib/bootstrap_form/form_group_builder.rb:89]: I like what you did here. I wonder how it interacts with https://github.com/bootstrap-ruby/bootstrap_form/blob/fe66580854058e243251ab6eabdbc42386ff68a5/lib/bootstrap_form/components/labels.rb#L28-L33. Is some of the code there now redundant?

**Review by lcreid — COMMENTED** [review]
Thanks very much for the contribution. I really appreciate anything anyone can contribute for accessibility. 

It looks good so far. I think we'll be able to merge this after you address the comments.

File: lib/bootstrap_form/form_group_builder.rb
Comment (CharlieWinkwaves) [reply lib/bootstrap_form/form_group_builder.rb:89]: I think you are right i changed it to only check for the options[:required] only one spec failed and that spec should not had have the required class anyway i think.

File: lib/bootstrap_form/inputs/check_box.rb
Comment (CharlieWinkwaves) [reply lib/bootstrap_form/inputs/check_box.rb:30]: You are correct i added a test for the checkbox, and also added the required class to the checkbox label

File: lib/bootstrap_form/components/labels.rb
Comment (lcreid) [inline lib/bootstrap_form/components/labels.rb:28]: Just looking at this change, it looks like a change to the behaviour of the software. Why don't we care about the `nil, :default` case anymore?

**Review by lcreid — COMMENTED** [review]
Thanks for adding the test. I have one more question. :-)

File: test/bootstrap_checkbox_test.rb
Comment (lcreid) [inline test/bootstrap_checkbox_test.rb:635]: This is a good test. Could you also add a test where the attribute is required by a validation, rather than by giving the `required` argument to the `check_box` helper?

File: test/bootstrap_form_group_test.rb
Comment (lcreid) [inline test/bootstrap_form_group_test.rb:321]: I believe this field is required. E-mail is required by the underlying model, so we want to make it a required field on the form, so this class would remain here, and you'd add the `aria-required` and `required` attributes to the `input`.

**Review by lcreid — COMMENTED** [review]
Thanks for your patience while waiting for my review. Let me know if my comments don't make sense.

I also got to thinking, if we're adding "required" to check boxes, is there a similar concept for radio button? What do you think? If you think it makes sense, let me know. If it's easy, maybe you can do it in this PR? If you think it merits its own PR, please create an issue so we can make sure it gets done.

Thanks again for your efforts to improve `bootstrap_form`!!

File: test/bootstrap_form_group_test.rb
Comment (CharlieWinkwaves) [reply test/bootstrap_form_group_test.rb:321]: You are right this was a bug in my code  I think I have fixed it

**Comment by CharlieWinkwaves** [comment]
> 

Yes maybe good to do it as well for radio buttons. I think i created a better solution to make sure the same check is done everywhere.

**Review by lcreid — APPROVED** [review]
Looks good! Thanks for your patience while I reviewed this, and thanks again for all the work you've done to make this a better gem.

**Pull request merged**
